### PR TITLE
Adds a reference point for number formats available.

### LINKED ISSUE #478: Enum or static number/date format ids and/or codes
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [ ] Bug
- [x] Feature

**What is the current behavior?**

I need to go to https://github.com/closedxml/closedxml/wiki/NumberFormatId-Lookup-Table to check which id corresponds to what format code. I was just wondering if you could you make this enum or some static instances with proper XML summary docs? Then I'd simply navigate to declarations and remain within an IDE. Moreover, your list is not complete as https://msdn.microsoft.com/en-us/library/documentformat.openxml.spreadsheet.numberingformat.aspx provides more options.

**What is the expected behavior or new feature?**

```csharp
range.Style.NumberFormat.NumberFormatId = NumberFormat._1; // or
range.Style.NumberFormat.NumberFormatId = NumberFormatIds._1 // or 
range.Style.NumberFormat.NumberFormatId = NumberFormats.ZeroZeroDotZero // etc.
```

### PR DESCRIPTION
**Pull request by wachulski**
#### What's this PR do?
Adds a reference point for number formats available.

No need to lookup ids/codes on the Internet. Closes #478.

#### Where should the reviewer start?

ClosedXML/Excel/Style/XLFormatIds.cs directly

#### How should this be manually tested?
n/a

#### Any background context you want to provide?
n/a

#### Questions:
- Is there a blog post? NO
- Does the knowledge base need an update? YES, in particular this https://github.com/closedxml/closedxml/wiki/NumberFormatId-Lookup-Table
- Does this add new (C#) dependencies? NO

- [ ] C# Code Review: @csreviewer
- [ ] Test Automation Review: @csreviewer

### CODE REVIEW

**Comment by igitur** [comment]
Isn't it better to use an `enum` for this?

**Comment by wachulski** [comment]
Well. With enum you need to cast to Int32 explicitly. But yes, this has more compile time benefits provided we used the enum as a NumberFormatId type instead of Int32. You to decide. I will still have to correct this as the class itself should be made static. So... ?

**Comment by igitur** [comment]
Oh, I see now you mentioned an `enum` approach in #478 . Yeah, I think an `enum` is better here. But let's consider a better naming convention rather than `F01`, `F02`, ... etc.

**Comment by wachulski** [comment]
Ok, I proposed more semantic approach. Categorized into number and datetime spaces. IMO should be more readable and user friendly to justify the change.

File: ClosedXML/Excel/Style/XLFormatIds.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
+﻿using System;
+
+namespace ClosedXML.Excel
+{
+    /// <summary>
+    /// Reference point of date/number formats available.
+    /// See more at: https://msdn.microsoft.com/en-us/library/documentformat.openxml.spreadsheet.numberingformat.aspx
+    /// </summary>
+    public static class XLFormat
```
Comment (igitur) [inline ClosedXML/Excel/Style/XLFormatIds.cs]: I think `XLPredefinedFormat` is clearer.

File: ClosedXML/ClosedXML.csproj [hunk 2]
```diff
@@ -83,6 +83,7 @@
     <Compile Include="Excel\Misc\XLCallbackAction.cs" />
     <Compile Include="Excel\Misc\XLReentrantEnumerableSet.cs" />
     <Compile Include="Excel\SaveOptions.cs" />
+    <Compile Include="Excel\Style\XLFormatIds.cs" />
```
Comment (igitur) [inline ClosedXML/ClosedXML.csproj]: Change to `XLPredefinedFormat.cs` in line with comment below.

**Comment by wachulski** [comment]
Ok, here you go.

**Pull request merged**